types/vue-ls/index.d.ts

```diff
@@ -0,0 +1,50 @@
+// Type definitions for vue-ls 3.2
+// Project: https://github.com/RobinCK/vue-ls#readme
+// Definitions by: Prateek Malhotra <https://github.com/someone1>
+//                 Oleg Spakov <https://github.com/lexasss>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.6
+
+import _Vue, { PluginFunction } from 'vue';
+
+declare module 'vue/types/vue' {
+    interface Vue {
+        $ls: WebStorage;
+    }
+
+    interface VueConstructor {
+        ls: WebStorage;
+    }
+}
+
+declare class WebStorage {
+    length: number;
+
+    constructor(storage: StorageTypes);
+    get(name: string, def?: any): any;
+    set(name: string, value: any, expire?: number): void;
+    key(index: number): string | null;
+    remove(name: string): void;
+    clear(): void;
+    on(name: string, callback: () => void): void;
+    off(name: string, callback: () => void): void;
+    setOptions(options?: StorageOptions): void;
+}
+
+export interface StorageOptions {
+    namespace?: string;
+    name?: string;
+    storage?: StorageTypes;
+}
+
+export enum StorageTypes {
+    Session = 'session',
+    Local = 'local',
+    Memory = 'memory',
+}
+
+declare const _default: {
+    install: PluginFunction<StorageOptions>;
+};
+
+export default _default;
```

Comment:
It seems like the [underlying library](https://unpkg.com/vue-ls@3.2.1/dist/vue-ls.js) is not using `export default`, so this .d.ts file should not either. Instead, you should follow the appropriate template from [the TypeScript handbook](http://www.typescriptlang.org/docs/handbook/declaration-files/templates.html). Note that if you want to use `import name from 'package';` in your consuming code, you can use the `esModuleInterop` compiler setting.

Reply:
You are right, I removed the default export. A new commit was added.

Reply:
@uniqueiniquity,
After `default` export was removed and this PR was merged, I figured out that my project where I use `vue-ls` receives a run-time error. After long research and trying out various module templates from  the TypeScript handbook, I found that only the original definition with `export default` works correctly. Neither `declare const VueStorage: ...` nor `declare global { const VueStorage: ... ; }` nor `export const VueStorage: ...` works here even if the TS compiler is happy about it: `VueStorage` is not a valid object when it comes into use.

Given this, and the fact that the `vue-ls` library exports VueStorage into a global space (`(window || self || global).VueStorage = ...`) and also returns this object from the library (`return VueStorage` at the end), is it OK if I open an new PR with `export default` as stated originally?

Reply:
That's disappointing that the `jsnext:main` and `main` versions of the library don't align - I assume you're using Rollup or Webpack or something of the sort, and getting the version that _does_ have a default export.

In this case, I think it is fine to open a new PR with `export default`. I would also suggest adding a comment explaining that this is because the assumed usage is in an environment that understands ES modules (i.e. Webpack or Rollup).

Reply:
Thanks! Indeed, `vue-ls` is meant to be used in VueJS projects with Webpack running behind (configured automatically by `vue-cli`).